solar_crypto/transactions/builder/vote.py

```diff
@@ -1,38 +1,60 @@
 import typing
+from math import trunc
 
-from solar_crypto.constants import TRANSACTION_VOTE
-from solar_crypto.identity.address import address_from_passphrase
+from solar_crypto.constants import SOLAR_TRANSACTION_VOTE, TRANSACTION_TYPE_GROUP
 from solar_crypto.transactions.builder.base import BaseTransactionBuilder
 
 
 class Vote(BaseTransactionBuilder):
 
-    transaction_type = TRANSACTION_VOTE
+    transaction_type = SOLAR_TRANSACTION_VOTE
+    typeGroup = TRANSACTION_TYPE_GROUP.SOLAR.value
 
-    def __init__(self, vote=None, fee=None):
-        """Create a second signature registration transaction
-
-        Args:
-            vote (str): address of a delegate you want to vote
-            fee (int, optional): fee used for the transaction (default is already set)
-        """
+    def __init__(self):
         super().__init__()
 
         self.transaction.asset["votes"] = []
-        if vote:
-            self.transaction.asset["votes"].append(vote)
-
-        if fee:
-            self.transaction.fee = fee
 
-    def set_votes(self, votes: typing.List[str]):
-        """Set votes/unvotes
+    def set_votes(
+        self,
+        votes: typing.Union[typing.List[str], typing.Dict[str, typing.Union[int, float]]] = dict,
+    ):
+        """Set votes
 
         Args:
-            votes (List[str]): list of votes
+            votes
         """
+        vote_object: typing.Dict[str, typing.Union[float, int]] = {}
+
+        if isinstance(votes, list):
+            vote_list = filter(lambda vote: not vote.startswith("-"), votes)
+            vote_list = list(map(lambda vote: vote[1:], vote_list))
+
+            if len(vote_list) == 0:
+                self.transaction.asset["votes"] = {}
+                return
+
+            weight = round((trunc((100 / len(vote_list)) * 100) / 100) * 100)
+            remainder = 10000
+
+            for vote in vote_list:
+                vote_object[vote] = weight / 100
+                remainder -= weight
+
+            for index in range(remainder):
+                key = list(vote_object.keys())[index]
+                vote_object[key] = round((vote_object[key] + 0.01) * 100) / 100
+
+            votes = vote_object
+
+        if votes:
+            nr_of_votes = len(votes.keys())
+            if nr_of_votes > 0:
+                votes = sort_votes(votes)
+
         self.transaction.asset["votes"] = votes
 
-    def sign(self, passphrase):
-        self.transaction.recipientId = address_from_passphrase(passphrase)
-        super().sign(passphrase)
+
+def sort_votes(votes: typing.Dict[str, typing.Union[float, int]]):
```

Comment:
Votes are sorted by descending weight and then by name (ascending).

e.g., `{"bbb": 50.5, "aaa": 25.25, "ccc": 25.25}`

> Reference: [https://github.com/Solar-network/core/blob/develop/packages/crypto/src/utils/sort-votes.ts](https://github.com/Solar-network/core/blob/cbb299b709de0b6785d879cdf8ccf0d1e568c2a7/packages/crypto/src/utils/sort-votes.ts)

Reply:
done

Reply:
Unless I'm reading it wrong, shouldn't it be:
```python
    sorted_votes = sorted(votes.items(), key=lambda vote: vote[1], reverse=True) # desc
    sorted_votes = sorted(votes.items(), key=lambda vote: vote[0]) # asc
```
Also,
1. Would line 2 re-sort the entire array, and
2. does Python support natural order by default?
    `{"a2a": 50.5, "a1a": 25.25, "a3a": 25.25}`

Reply:
This whole sort function is wrong. I'll rethink this.

Reply:
I've pushed a changed in which should fix it: https://github.com/Solar-network/python-crypto/pull/30/commits/55414ce0526df9f70070d521efc7fecd4eee42a3